### PR TITLE
MAINT, DOC: add eyetracking & Dipole convention for loc array to Info class API

### PR DESCRIPTION
**Pull request by scott-huberty**
Originally brought up in https://github.com/mne-tools/mne-python/pull/11798#discussion_r1337919770

@larsoner / @drammock  / @britta-wstnr , can you read the text I added, and let me know if it is clear/sensible to you?

### CODE REVIEW

File: mne/_fiff/meas_info.py [hunk 1]
```diff
@@ -1189,7 +1190,10 @@ class Info(dict, SetChannelsMixin, MontageMixin, ContainsMixin):
             normal given by a 3x3 rotation matrix. For EEG this is the
             position followed by reference position (with 6 unused).
             The values are specified in device coordinates for MEG and in
-            head coordinates for EEG channels, respectively.
+            head coordinates for EEG channels, respectively. For eye-tracking channels,
+            index 3 stores information about which eye was tracked (-1 for left, 1 for
+            right), and index 4 stores information about the the axis of coordinate
+            data (-1 for x-coordinate data, 1 for y-coordinate data).
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mne/_fiff/meas_info.py:1196]: Really this whole section begs to be put in a bulleted or definition list, like:
```
Channel location information. The first three elements always store the
nominal channel position. The remaining 9 elements store different
information based on the channel type:

EEG
    Elements ``[3:6]`` contain the reference channel position.
MEG
    Remaining elements contain the EX, EY, and EZ normals (columns) of
    the coil rotation/orientation matrix.
...
```
Not 100% sure if the definition list above would look better or a bulleted list, but my guess is def list. I'd `make html-noplot` both versions to see which rendered better.

File: mne/_fiff/meas_info.py [hunk 2]
```diff
@@ -998,6 +998,7 @@ def _check_helium_info(helium_info):
     return helium_info
 
 
+# TODO: Add fNIRS convention to loc
```
Comment (larsoner) [inline mne/_fiff/meas_info.py:1001]: Digging through I found that `dipole` is also not documented.
```suggestion
# TODO: Add fNIRS and dipole convention to loc
```
If you feel like documenting it, `[3:6]` stores the dipole orientation.

**Comment by scott-huberty** [comment]
[Here is new rendering of  the docstring](https://output.circle-artifacts.com/output/job/88426560-ed89-4d96-8445-d71b20bb61e2/artifacts/0/html/generated/mne.Info.html):

File: mne/_fiff/meas_info.py [hunk 3]
```diff
@@ -1185,11 +1186,23 @@ class Info(dict, SetChannelsMixin, MontageMixin, ContainsMixin):
...
+            EEG
+                Elements ``[3:6]`` contain the reference channel position.
+            Eyetrack
+                Element ``[3]`` contains information about which eye was tracked
+                (-1 for left, 1 for right), and element ``[4]`` contains information
+                about the the axis of coordinate data (-1 for x-coordinate data, 1 for
+                y-coordinate data).
+            Dipole
+                Elements ``[3:6]`` contain dipole orientation information.
+
```
Comment (larsoner) [inline mne/_fiff/meas_info.py:1205]: 90% sure you don't need this newline (and numpydoc might complain someday) -- let's find out the hard way
```suggestion
```

**Review by larsoner — APPROVED** [review]
Marking for merge when green, thanks @scott-huberty !

**Pull request merged**